api/client.go

```diff
@@ -1145,6 +1151,103 @@ func RequireState(states ...string) RequestCallback {
 	}
 }
 
+// CompareStates returns 1 if s1 is newer or identical, -1 if s1 is older, and 0
+// if neither s1 or s2 is strictly greater. An error is returned if s1 or s2
+// are invalid or from different clusters.
+func CompareStates(s1, s2 string) (int, error) {
```

Comment:
```suggestion
func CompareReplicationStates(s1, s2 string) (int, error) {
```

States seems quite generic to me, and for a public method in our `api` package, won't give developers much intuition for what these methods are to do with unless they're already very familiar with Vault's internals. Is it worth making the method names a little more specific to help? Would love a second opinion on the suggestion though.

Reply:
I agree with Tom, and if you accept this please don't forget to change the documentation comment above it.

Reply:
Thanks for the suggestion! Updated the method name